### PR TITLE
Vulkan: Refactor renderpass handling

### PR DESCRIPTION
**Pull request by hrydgard**
Hide all the renderpass compatibility management inside VulkanRenderManager / QueueRunner.

This is needed before I can properly finish #15944 , as it can just piggyback onto the renderpass management here.

Also, this will enable fully depth-less renderpasses and other fun stuff later (which will increase GPU performance on mobile).

Draft, still needs some work around shader caching.

### CODE REVIEW

**Comment by hrydgard** [comment]
Ah, the "bugs" I was experiencing were simply caused by render pass merging, gotta handle that correctly. It might mean that we don't know until final draw time what render pass type a pass is, so we might sit there with uncompiled pipelines and have to compile them just-in-time, which isn't great for stutter. Maybe some kind of middle pass can solve it. I'll see...

**Comment by hrydgard** [comment]
Alright, got the whole thing working on another branch, then extracted all the non-blending-related refactoring here. Turned out to be quite a bit.

**Comment by hrydgard** [comment]
There seems to be a problem on Android, switching back to draft.

**Comment by hrydgard** [comment]
Never mind, fixed!

File: GPU/Common/FragmentShaderGenerator.h [hunk 1]
```diff
@@ -39,4 +39,9 @@ struct FShaderID;
 // For stencil upload
 #define CONST_PS_STENCILVALUE 11
 
-bool GenerateFragmentShader(const FShaderID &id, char *buffer, const ShaderLanguageDesc &compat, Draw::Bugs bugs, uint64_t *uniformMask, std::string *errorString);
+// Can technically be deduced from the fragment shader ID, but this is safer.
+enum FragmentShaderFlags {
+	FS_FLAG_INPUT_ATTACHMENT,
```
Comment (unknownbrackets) [inline GPU/Common/FragmentShaderGenerator.h:44]: I haven't looked at the other pull yet, but isn't this 0?  Shouldn't we have a `= 1` or NONE?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]

File: GPU/Common/FragmentShaderGenerator.h (hunk 1 above)
Comment (hrydgard) [reply GPU/Common/FragmentShaderGenerator.h:44]: Ah, this is fixed in the other PR that actually makes use of it. This is just a buildfix from a rebase, trying to tease the code apart...

**Pull request merged**